Library/Homebrew/exceptions.rb

```diff
@@ -582,16 +582,27 @@ def initialize(cmd, status:, output: nil, secrets: [])
     exitstatus = case status
     when Integer
       status
+    when Hash
+      status["exitstatus"]
     else
       status&.exitstatus
     end
 
+    termsig = case status
+    when Integer
```

Comment:
```suggestion
    when Integer, NilClass
```

Reply:
Looking at this again, I think an early `raise` for a nil `status` seems appropriate.